crates/aptos/src/node/mod.rs

```diff
@@ -1486,65 +1488,27 @@ impl CliCommand<()> for AnalyzeValidatorPerformance {
 ///
 /// Enables users to load from a backup to catch their node's DB up to a known state.
 #[derive(Parser)]
-pub struct BootstrapDbFromBackup {
-    /// Config file for the source backup
-    ///
-    /// This file configures if we should use local files or cloud storage, and how to access
-    /// the backup.
-    #[clap(long, parse(from_os_str))]
-    config_path: PathBuf,
-
-    /// Target database directory
-    ///
-    /// The directory to create the AptosDB with snapshots and transactions from the backup.
-    /// The data folder can later be used to start an Aptos node. e.g. /opt/aptos/data/db
-    #[clap(long = "target-db-dir", parse(from_os_str))]
-    pub db_dir: PathBuf,
-
+pub struct BootstrapDb {
     #[clap(flatten)]
-    pub metadata_cache_opt: MetadataCacheOpt,
-
+    storage: DBToolStorageOpt,
     #[clap(flatten)]
-    pub concurrent_downloads: ConcurrentDownloadsOpt,
-
+    opt: RestoreCoordinatorOpt,
     #[clap(flatten)]
-    pub replay_concurrency_level: ReplayConcurrencyLevelOpt,
+    global: GlobalRestoreOpt,
 }
 
 #[async_trait]
-impl CliCommand<()> for BootstrapDbFromBackup {
+impl CliCommand<()> for BootstrapDb {
     fn command_name(&self) -> &'static str {
-        "BootstrapDbFromBackup"
+        "BootstrapDb"
     }
 
     async fn execute(self) -> CliTypedResult<()> {
-        let opt = RestoreCoordinatorOpt {
-            metadata_cache_opt: self.metadata_cache_opt,
-            replay_all: false,
-            ledger_history_start_version: None,
-            skip_epoch_endings: false,
-        };
-        let global_opt = GlobalRestoreOpt {
-            dry_run: false,
-            db_dir: Some(self.db_dir),
-            target_version: None,
-            trusted_waypoints: Default::default(),
-            rocksdb_opt: RocksdbOpt::default(),
-            concurrent_downloads: self.concurrent_downloads,
-            replay_concurrency_level: self.replay_concurrency_level,
-        }
-        .try_into()?;
-        let storage = Arc::new(CommandAdapter::new(
-            CommandAdapterConfig::load_from_file(&self.config_path).await?,
-        ));
-
-        // hack: get around this error, related to use of `async_trait`:
-        //   error: higher-ranked lifetime error
-        //   ...
-        //   = note: could not prove for<'r, 's> Pin<Box<impl futures::Future<Output = std::result::Result<(), CliError>>>>: CoerceUnsized<Pin<Box<(dyn futures::Future<Output = std::result::Result<(), CliError>> + std::marker::Send + 's)>>>
+        let storage = self.storage.init_storage().await?;
         tokio::task::spawn_blocking(|| {
             let runtime = tokio::runtime::Runtime::new().unwrap();
-            runtime.block_on(RestoreCoordinator::new(opt, global_opt, storage).run())
+            runtime
+                .block_on(RestoreCoordinator::new(self.opt, self.global.try_into()?, storage).run())
```

Comment:
I wonder if this is still necessary, vs just awaiting the inner function directly? If it _is_ still necessary, can you please leave that comment from before explaining the hack.

Reply:
still necessary. Otherwise same compiler error with just inner function. added back the comment